sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/BenchmarkRequestSubscriber.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.BaseSubscriber;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class BenchmarkRequestSubscriber<T> extends BaseSubscriber<T> {
```

Comment:
This is used only in CTL so `BenchmarkRequestSubscriber` -> `CTLBenchmarkRequestSubscriber`
or perhaps create a sub package for CTL and put CTL related code in that package?

`com.azure.cosmos.benchmark.ctl` ? thought Naveen?

Reply:
Moved AsyncCtlWorkload to com.azure.cosmos.benchmark.ctl. BenchmarkRequestSubscriber  is not just ctl related  and can be reused in other places. Could have done the refactoring(it is non minimal) , but it would have increased the scope of this PR.  Will move all these common classes to common packages in future work.